webapp/packages/core-ui/src/Tabs/useTabPersistence.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { useCallback, useMemo, useRef } from 'react';
+import { untracked, action } from 'mobx';
+
+import { useUserData, useExecutor } from '@cloudbeaver/core-blocks';
+import { isArraysEqual } from '@cloudbeaver/core-utils';
+
+import type { TabsContainer } from './TabsContainer/TabsContainer.js';
+
+interface ITabPersistenceState {
+  selectedTabId: string | undefined;
+}
+
+export function useTabPersistence(panelId: string, container: TabsContainer) {
+  const state = useUserData<ITabPersistenceState>(panelId, () => ({ selectedTabId: undefined }));
+
+  const tabs = container.getIdList();
+  const prevTabs = useRef<string[]>(tabs);
+  const equal = isArraysEqual(prevTabs.current, tabs);
+
+  const selectTab = useCallback(
+    (tabId: string) => {
+      state.selectedTabId = tabId;
+    },
+    [state],
+  );
+
+  useExecutor({
+    executor: container.onTabSelect,
+    handlers: [
+      function selectTabHandler(tabId) {
+        selectTab(tabId);
+      },
+    ],
+  });
+
+  untracked(
+    action(() => {
```

Comment:
in mobx `action` is untracked by default, so you can use `runInAction` without `untracked`

Reply:
I took the existing logic and reuse it. I will try to adjust it and test